testing/scenario_app/ios/Scenarios/ScenariosTests/AppLifecycleTests.m

```diff
@@ -6,6 +6,28 @@
 #import <XCTest/XCTest.h>
 #import "ScreenBeforeFlutter.h"
 
+@interface XCAppLifecycleTestExpectation : XCTestExpectation
+
+- (instancetype)initForLifecycle:(NSString*)expectedLifecycle forStep:(NSString*)step;
+@property(nonatomic, readonly) NSString* expectedLifecycle;
+
+@end
+
+@implementation XCAppLifecycleTestExpectation
+
+@synthesize expectedLifecycle = _expectedLifecycle;
+- (instancetype)initForLifecycle:(NSString*)expectedLifecycle forStep:(NSString*)step {
+  // The step is here because the callbacks into the handler which checks these expectations isn't
+  // synchronous with the executions in the test, so it's hard to find the cause in the test
+  // otherwise.
+  self = [self initWithDescription:[NSString stringWithFormat:@"Expected state %@ during step %@",
```

Comment:
`[super initWithDescription:]` not self.

Reply:
done